L3 ht

taras-petushkov opened the pull request:
see commit messages for more details

taras-petushkov committed:
Lesson 3| review all css classes after BEM intro, split all big css to the small ones

taras-petushkov committed:
Lesson 3| move all templates to the separate files, compile them manually and include all in html. later gulp will compile and join all templates in one js

taras-petushkov committed:
Lesson 3| move all partial registrations to the corresponding folders, unify scripts imported by two pages. next step will be playing with gulp

taras-petushkov committed:
Lesson 3| setup gulp build (initial version)

taras-petushkov committed:
Lesson 3| setup gulp watch task

taras-petushkov committed:
Lesson 3| setup gulp style task

taras-petushkov committed:
Lesson 3| update gulp tasks to minify html and build source maps

taras-petushkov committed:
Lesson 3| checking style only for modified files

taras-petushkov committed:
Lesson 3| small rearrangement inside gulp file

taras-petushkov committed:
Lesson 3| few more gulp tasks - autoprefixer and csscomb

taras-petushkov committed:
Lesson 3| integrate livereload with watch

taras-petushkov committed:
Lesson 3| attempt to fix broken markdown...

taras-petushkov committed:
Lesson 3| proper js path in gulp file js

puzankov commented on gulpfile.js:
```diff
@@ -0,0 +1,171 @@
+        .pipe(sourcemaps.init())
+        .pipe(autoprefixer())
+        .pipe(gulpif(isProd, cssnano()))
+        .pipe(sourcemaps.write(cssMapDir))
```
Delete sourcemap from prod

taras-petushkov replied:
done. my understanding of sourcemaps was wrong

puzankov commented on gulpfile.js:
```diff
@@ -0,0 +1,171 @@
+});
+
+function gitmodified() {
+    return gitmodified0(['added', 'modified', 'untracked', 'renamed', 'copied', 'updated but unmerged']);
```
what is 0

taras-petushkov replied:
renamed to gitmodifiedOrigin

puzankov commented on gulpfile.js:
```diff
@@ -0,0 +1,171 @@
+        .pipe(jscs({ fix: true }))
+        .pipe(gulp.dest(srcDir));
+});
+
```
check configs usage

taras-petushkov replied:
checked by modifying configurations and checking gulp messages. everything works as expected

taras-petushkov committed:
Lesson 3| fixes after mentor review:
+ move all directory constants to existing object path
+ rename variable gitmodified0-> gitmodifiedOrigin
+ proper source map generation
+ enable W069 check for jshint

puzankov commented:
+1
